annotate.el

```diff
@@ -896,5 +936,80 @@ file."
+                                       annotate-ellipse-text-marker)
+                             text)))
+              (wrap      (text)
+                         (concat "\"" text "\""))
+              (insert-item-summary (snippet-text button-text)
+                                   (insert annotate-summary-list-prefix-snippet)
+                                   (insert (ellipsize (wrap snippet-text)
+                                                      annotate-summary-list-prefix-snippet))
+                                   (insert "\n")
+                                   (insert annotate-summary-list-prefix)
+                                   (insert-button (propertize (ellipsize button-text
+                                                                         annotate-summary-list-prefix)
+                                                              'face
+                                                              'bold)
+                                                  'file   filename
+                                                  'go-to  annotation-begin
+                                                  'action 'annotate-summary-button-pressed
+                                                  'type   'annotate-summary-button)
+                                   (insert "\n\n"))
+              (build-snippet ()
+                             (with-temp-buffer
+                               (insert-file-contents filename
+                                                     nil
+                                                     (1- annotation-begin)
+                                                     (1- annotation-end))
+                               (buffer-string))))
+    (with-current-buffer-window
+     "*annotations*" nil nil
+     (display-buffer "*annotations*")
+     (select-window (get-buffer-window "*annotations*" t))
+     (outline-mode)
+     (use-local-map nil)
+     (local-set-key "q" (lambda ()
+                          (interactive)
+                          (kill-buffer "*annotations*")))
+     (let ((dump (annotate-load-annotation-data)))
+       (dolist (annotation dump)
+         (let ((all-annotations (annotate-annotations-from-dump annotation))
+               (filename        (annotate-filename-from-dump annotation)))
+           (when (not (null all-annotations))
+             (insert (format (concat annotate-summary-list-prefix-file "%s\n\n")
+                             filename))
+             (dolist (annotation-field all-annotations)
+               (let* ((button-text      (format "%s"
+                                                (annotate-text-of-annotation annotation-field)))
+                      (annotation-begin (annotate-beginning-of-annotation annotation-field))
+                      (annotation-end   (annotate-ending-of-annotation    annotation-field))
+                      (snippet-text     (build-snippet)))
```

Comment:
In general, I prefer functions to have explicit arguments instead of referring to local dynamic bindings. Is there a specific reason why you implemented `build-snippet` this way?

Reply:
Yes, this is confusing at best, fixed!